reladomographql/src/test/java/com/gs/reladomo/graphql/SchemaProviderTest.java

```diff
@@ -0,0 +1,206 @@
+/*
+ Copyright 2019 Goldman Sachs.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ */
+
+package com.gs.reladomo.graphql;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.gs.fw.common.mithra.test.MithraTestResource;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.schema.GraphQLSchema;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+public class SchemaProviderTest extends GraphqlTestBase {
+    @Test
+    public void findOne() throws Exception {
+        assertQuery(
+                "{ productById ( productId: 1 ) { productId, productCode } }",
+                "{\"productById\":{\"productId\":1,\"productCode\":\"AA\"}}");
+    }
+
+    @Test
+    public void findManyWithDeepFetch() throws Exception {
+        assertQuery(
+                "{ orders (findMany: {orderId: {in: [2,3,4,10]} }) { orderId description items { state } } }",
+                "{\"orders\":[{\"orderId\":2,\"description\":\"Second order\",\"items\":[{\"state\":\"InProgress\"},{\"state\":\"InProgress\"},{\"state\":\"InProgress\"}]}," +
+                        "{\"orderId\":3,\"description\":\"Third order\",\"items\":[]}," +
+                        "{\"orderId\":4,\"description\":\"Fourth order, different user\",\"items\":[]}]}\n");
+    }
```

Comment:
I'm curious why the NameById queries look so different than the findMany queries.

Why did you go with:
```graphql
{ productById ( productId: 1 ) { productId, productCode } }
```

And not something like
```graphql
{ product ( byId: {productId: 1} ) { productId, productCode } }
```